Lib/pdb.py

```diff
@@ -1073,6 +1093,35 @@ def _select_frame(self, number):
         self.print_stack_entry(self.stack[self.curindex])
         self.lineno = None
 
+    def do_exceptions(self, arg):
+        """exceptions [number]
+
+        List or change current exception in an exception chain.
+
+        Without arguments, list all the current exception in the exception
+        chain. Exceptions will be numbered, with the current exception indicated
+        with an arrow.
+
+        If given an integer as argument, switch to the exception at that index.
+        """
+        if not arg:
+            nwid = len(str(len(self._chained_exceptions) - 1))
```

Comment:
I think we should just use 3 and somehow limit the number of exceptions in a chain you browse to 999. Truncate (with some msg) if there are more or something like that.

Which reminds me that we also need to handle cycles in the chain. Look at the traceback module to see how it's done there.

Reply:
Ok, hardcoded to 3 digits (999) set on a private class variable of Pdb just in case.

At startup Pm stops after 999 exception and print a message that only the first 999 will be browsable with `exceptions`. I was not sure it was worth to make that lazy until called for in `do_exceptions`.

While I was also dealing with cycles, I added support for exceptions groups as well and added test with various exceptions, groups, cause, context...